Build failure: python3Packages.hickle

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. go to the root of nixpkgs repo master
2. run `nix-build -A python3Packages.hickle -I=.`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.lossy.network/build/2944421

### Relevant log output

```console
>               content = np.array(content, copy=False, dtype=str).tolist()
E               ValueError: Unable to avoid copy while creating an array as requested.
E               If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
E               For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.
```

### Additional context

The tests are failing probably due to switch to numpy 2.x, there is a [PR upstream that fixes it](https://github.com/telegraphic/hickle/pull/186)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.7, NixOS, 24.11 (Vicuna), 24.11.20250318.da04445`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/store/s3bhg6p26cqi1hm2lamjjdwaiq49g2hn-source`

### Notify maintainers

@bcdarwin

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/hickle/default.nix
+++ b/pkgs/development/python-modules/hickle/default.nix
@@ -2,6 +2,7 @@
   lib,
   buildPythonPackage,
   fetchPypi,
+  fetchpatch,
   pythonOlder,
   h5py,
   numpy,
@@ -25,6 +26,15 @@ buildPythonPackage rec {
     inherit pname version;
     hash = "sha256-An5RzK0nnRaBI6JEUl5shLrA22RgWzEbC9NJiRvgxT4=";
   };
+
+  patches = [
+    # fixes support for numpy 2.x, the PR is not yet merged https://github.com/telegraphic/hickle/pull/186
+    # FIXME: Remove this patch when the numpy 2.x support arrives
+    (fetchpatch {
+      url = "https://github.com/cjwatson/hickle/commit/246d8e82c805e2e49ea0abd39abc9b2d800bde59.patch";
+      hash = "sha256-IEVw2K7S1nCkzgn9q0xghm4brfXcallNjzXpt2cRq1M=";
+    })
+  ];
 
   build-system = [ setuptools ];
 

